### DISCUSSION TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'writeText')

### DISCUSSION THREAD

**Discussion by saleh-movahedi** [filamentphp/filament#9282] [Help]
I have a text component in infolist 
```
  \Filament\Infolists\Components\TextEntry::make('email')
      ->label('ایمیل')
      ->copyable()

```
when I click the email field it doesn't work this error is being showed in console:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'writeText')
    at [Alpine] 
                window.navigator.clipboard.writeText('test_mail@example.com')

```

**Comment by sinnbeck** [comment]
Which browser is that? The support for clipboard is pretty good
https://caniuse.com/?search=navigator.clipboard

**Reply by saleh-movahedi** [reply]
chrome 117 on linux ubuntu, wayland
the browser is up to date.
I don't understand. it works sometimes, without any changes in code!

**Reply by sinnbeck** [reply]
When it fails try typing window.navigator.clipboard in the browser console to see if it's gone/broken

**Comment by malzariey** [comment]
> I have a text component in infolist
> 
> ```
>   \Filament\Infolists\Components\TextEntry::make('email')
>       ->label('ایمیل')
>       ->copyable()
> ```
> 
> when I click the email field it doesn't work this error is being showed in console:
> 
> ```
> Uncaught TypeError: Cannot read properties of undefined (reading 'writeText')
>     at [Alpine] 
>                 window.navigator.clipboard.writeText('test_mail@example.com')
> ```

try using HTTPS.
